### PR TITLE
[Pygen]: Add support for illegal and load/store instructions.

### PR DESCRIPTION
**Pull request by aneels3**
This PR adds the support for the following test:

- riscv_illegal_instr_test
- Directed load/store stream
1. riscv_load_store_rand_instr_stream
2. riscv_load_store_hazard_instr_stream
3. riscv_load_store_stress_instr_stream
4. riscv_single_load_store_instr_stream

### CODE REVIEW

**Comment by pvipsyash** [comment]
@googlebot I consent.

File: pygen/pygen_src/isa/riscv_compressed_instr.py [hunk 1]
```diff
@@ -31,54 +31,54 @@ def __init__(self):
     @vsc.constraint
     def rvc_csr_c(self):
         # Registers specified by the three-bit rs1’, rs2’, and rd’
-        with vsc.implies(self.format.inside(vsc.rangelist(riscv_instr_format_t.CIW_FORMAT,
+        with vsc.if_then(self.format.inside(vsc.rangelist(riscv_instr_format_t.CIW_FORMAT,
```
Comment (udinator) [inline pygen/pygen_src/isa/riscv_compressed_instr.py:34]: just wondering, was there some issue with using the `implies` constraints?

File: pygen/pygen_src/riscv_illegal_instr.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+        self.compressed = vsc.rand_bit_t(1)
+        self.func3 = vsc.rand_bit_t(3)
+        self.func7 = vsc.rand_bit_t(7)
+        self.has_func3 = vsc.rand_bit_t(1)
+        self.has_func7 = vsc.rand_bit_t(1)
+        self.c_op = vsc.rand_bit_t(2)
+        self.c_msb = vsc.rand_bit_t(3)
+        self.csrs = []
+        # Default legal self.opcode for RV32I instructions
+        self.legal_opcode = vsc.list_t(vsc.bit_t(7))
```
Comment (udinator) [inline pygen/pygen_src/riscv_illegal_instr.py:74]: maybe I'm not too familiar with the PyVCS syntax...is there a way to do a one-line list initialization here?

**Review by udinator — APPROVED** [review]
lgtm in general, with a few nit comments

File: pygen/pygen_src/isa/riscv_compressed_instr.py (hunk 1 above)
Comment (aneels3) [reply pygen/pygen_src/isa/riscv_compressed_instr.py:34]: Hi @udinator 
No, back then there was some issue with the if_then constructs. I have changed this Just to be consistent with the if and else statement.

File: pygen/pygen_src/riscv_illegal_instr.py (hunk 2 above)
Comment (aneels3) [reply pygen/pygen_src/riscv_illegal_instr.py:74]: one needs to define the type first to assign the corresponding values.
There is no way to do this in one line.

**Comment by aneels3** [comment]
hi @udinator 
I have melded the previous commits into one. For me, there are no conflicts with the base branch.
Can you check once at your end?

**Comment by udinator** [comment]
Merging the commits seems to have resolved the issue, thanks!

**Pull request merged**